src/rapier_body_utils_2d.cpp

```diff
@@ -99,7 +90,9 @@ bool RapierBodyUtils2D::body_motion_recover(
 				Vector2 b(contact.point2.x, contact.point2.y);
 
 				// Compute plane on b towards a.
-				Vector2 n = (a - b).normalized();
+				Vector2 n = Vector2(contact.normal1.x, contact.normal1.y);
```

Comment:
I find a-b clearer, are you sure that it's normal1 that should be used and not 2?

Reply:
A-b is cleaner, but normal is already computed, so its more efficient like this. Y, its normal 1.